platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/sync/SyncCaller.java

```diff
@@ -70,27 +80,42 @@ public class SyncCaller implements Runnable {
      * The platform instantiates this, and gives it the self ID number, plus other info that will be useful to it. The
      * platform can then create a thread to call run(). It will then repeatedly initiate syncs with others.
      *
-     * @param platform     the platform using this caller
-     * @param addressBook  the address book listing who can be called
-     * @param selfId       the ID number for self /** ID number for this caller thread (0 is the first created by this
-     *                     platform, 1 next, etc)
-     * @param callerNumber 0 for the first caller thread created by this platform, 1 for the next, etc
+     * @param platformContext          the platform context
+     * @param addressBook              the address book listing who can be called
+     * @param selfId                   the ID number for self /** ID number for this caller thread (0 is the first
+     *                                 created by this platform, 1 next, etc)
+     * @param callerNumber             0 for the first caller thread created by this platform, 1 for the next, etc
+     * @param simultaneousSyncThrottle the throttle to use to limit the number of simultaneous syncs
+     * @param syncManager              the sync manager
+     * @param sharedConnectionLocks    the shared connection locks
+     * @param eventTaskCreator         responsible for scheduling the creation of events
      */
     public SyncCaller(
-            final SwirldsPlatform platform,
-            final AddressBook addressBook,
-            final NodeId selfId,
+            @NonNull final PlatformContext platformContext,
+            @NonNull final AddressBook addressBook,
+            @NonNull final NodeId selfId,
             final int callerNumber,
-            final ReconnectHelper reconnectHelper,
-            final SignedStateValidator signedStateValidator) {
-        this.platform = platform;
-        this.addressBook = addressBook;
-        this.selfId = selfId;
+            @NonNull final ReconnectHelper reconnectHelper,
+            @NonNull final SignedStateValidator signedStateValidator,
+            @NonNull final SimultaneousSyncThrottle simultaneousSyncThrottle,
+            @NonNull final SyncManagerImpl syncManager,
+            @NonNull final SharedConnectionLocks sharedConnectionLocks,
+            @NonNull final EventTaskCreator eventTaskCreator,
+            @NonNull final Runnable updatePlatformStatus,
```

Comment:
missing javadocs for some of there parameters

Reply:
added